pdm/installers/installers.py

```diff
@@ -25,6 +25,86 @@
 
     from installer.destinations import Scheme
 
+    from pdm.models.environment import Environment
+
+
+@lru_cache()
+def _is_python_package(root: str | Path) -> bool:
+    for child in Path(root).iterdir():
+        if (
+            child.is_file()
+            and child.suffix in (".py", ".pyc", ".pyo", ".pyd")
```

Comment:
I wonder if `.so` files should be checked as well.

Reply:
I've thought of that, but it was not that easy to tell a `.so` is a python extension module or not. So this case can be ignored given how rare it is to put a `.so` inside a PEP 420 namespace package.

Reply:
I've just encountered that case :joy: That's actually why I'm commenting this :joy:
But yeah, more than the extension, the whole file name would need to be checked against some regular expression like `^(?P<module_name>\w+).(?P<build_info>>.+).so$` (from my limited knowledge)😕